mods/auto-hide-desktop-after-inactivity.wh.cpp

```diff
@@ -0,0 +1,327 @@
+// ==WindhawkMod==
+// @id              auto-hide-desktop-after-inactivity
+// @name            Auto Hide Desktop After Inactivity
+// @description     Automatically hides all windows and shows the desktop after a period of user inactivity
+// @version         0.5
+// @author          alekseev2014s
+// @github          https://github.com/Alekseev2014s
+// @include         explorer.exe
+// @compilerOptions -lcomdlg32
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Auto Hide Desktop After Inactivity
+
+This mod automatically minimizes all open windows and hides the desktop
+after a period of user inactivity. It's useful for shared environments,
+kiosks, or simply to reduce visual clutter after you're away from your PC.
+
+## Features
+- **Inactivity timeout**: Automatically hide the desktop after no input for a configurable number of seconds.
+- **Lock screen timeout**: Optionally lock the workstation after a longer period of inactivity.
+- **Fullscreen app detection**: Prevents the desktop from being hidden if a fullscreen application is active.
+- **Process exclusions**: You can specify a list of executable names (like `vlc.exe`) to ignore during monitoring.
+
+## How it works
+- When the system is idle for the specified timeout, the mod simulates a `Win+D` keystroke to hide all windows and show the desktop.
+- If configured, after further inactivity, it automatically locks the workstation.
+- User input such as mouse movement or keyboard activity restores the previous window state.
+- Foreground fullscreen applications and excluded processes will prevent the desktop from being hidden.
+
+## Getting started
+1. Install the mod in Windhawk.
+2. Adjust the settings for:
+   - Timeout (in seconds)
+   - Lock timeout (optional)
+   - Excluded processes
+   - Fullscreen app handling
+3. Let the system idle and watch the desktop auto-hide behavior in action.
+
+## Example use cases
+- **Privacy**: Automatically hides sensitive windows when away.
+- **Kiosk mode**: Resets to desktop after inactivity.
+- **Clutter reduction**: Keeps your desktop clean when you're not using the PC.
+
+## Troubleshooting
+- If the desktop is not hidden as expected, check that no fullscreen app or excluded process is active.
+- To ensure it only runs once, the mod uses a global mutex to avoid interfering with other `explorer.exe` instances.
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- timeoutSeconds: 300
+  $name: Inactivity timeout (seconds)
+  $description: The number of seconds of inactivity before hiding the desktop.
+
+- lockTimeoutSeconds: 600
+  $name: Lock screen timeout (seconds)
+  $description: The number of seconds of inactivity before locking the workstation.
+
+- excludeProcesses: ""
+  $name: Excluded processes
+  $description: Semicolon-separated list of executable names to exclude from triggering desktop show (e.g. "vlc.exe;notepad.exe")
```

Comment:
Why not use an array instead? For example:

https://github.com/ramensoftware/windhawk-mods/blob/9d7df8f53b5d834fbed6034c7c7a2dd03033f66a/mods/taskbar-grouping.wh.cpp#L93-L94

Reply:
Update the description, it's no longer a semicolon-separated list.

Reply:
In the most recent update, you changed some other description, but not this one (search for "Semicolon" in the code).